### ISSUE TITLE
hwsim: re-organize to be under programs

### ISSUE THREAD

**Issue by rli9** [intel/lkp-tests#234]
Refer to d7206025f284 as example. Also add meta.yaml for hwsim.

### Acceptance Test
* The components of test is moved to programs
* Refer to lkp-tests/docker/README.md, docker build needs to be done for only once to create a base docker image, then follow the “Run one atomic job” section to go through “lkp split-job”, “lkp install”, “lkp run”, “lkp result” steps.
* Once “lkp result” command can show a good result path with expected result files in it, we judge this as a good re-organization and accept it. e.g.
```
root@lkp-docker:/lkp/lkp-tests# lkp rt hackbench
/lkp/result/hackbench/pipe-8-process-1600%/lkp-docker/debian/defconfig/gcc-8/5.15.0-60-generic/0/

[root@lkp-docker:/lkp/result/hackbench/pipe-8-process-1600%/lkp-docker/debian/defconfig/gcc-8/5.15.0-60-generic/0#](mailto:root@lkp-docker:/lkp/result/hackbench/pipe-8-process-1600%25/lkp-docker/debian/defconfig/gcc-8/5.15.0-60-generic/0#) ls
bdi_dev_mapping  hackbench.json       matrix.csv       post-run.perf-profile  slabinfo.json.gz  uptime
bg-procs-debug   hackbench.time       matrix.json.gz   post-run.perf-sched    softirqs.gz       uptime.json
cpufreq-stats    hackbench.time.json  meminfo.gz       proc-stat.gz           softirqs.json     vmstat
cpuidle          heartbeat            meminfo.json.gz  proc-vmstat.gz         stats.csv         vmstat.json.gz
cpuidle.json     interrupts.gz        mpstat           proc-vmstat.json.gz    stats.json
diskstats.gz     interrupts.json      mpstat.json.gz   program_list           time
diskstats.json   job.sh               mpstat_update    reproduce.sh           time.json
hackbench        job.yaml             perf-stat.gz     slabinfo               turbostat
```

**Comment by fongjiantan** [comment]
LKP_Team3 working on this

**Comment by j-chiam** [comment]
On this issue, seems like cloning `git://w1.fi/srv/git/hostap.git` repository is unsuccessful. 
We've tried using this repo: `http://w1.fi/hostap.git` instead and it works. 
Is it preferred to fix it this issue, or a separate one? 

Besides, the modprobe steps for hwsim fails in docker env, as docker does not have access to /lib/modules. 
Please advise on how to verify the changes in this case, thank you!

**Comment by rli9** [comment]
> Is it preferred to fix it in this issue, or a separate one?

You can fix it in a new issue, like report a bug for this, then submit PR to fix.

> the modprobe steps for hwsim fails in docker env, as docker does not have access to /lib/modules

you can give a try with --privileged option as below. But if it still doesn't work, it's ok to not fully test it.
```
	# Add --privileged option to allow privileged access like dmesg, sysctl. Use
	# this with caution.
	docker run -d -h lkp-docker --name lkp-docker \
	           -v /home/$USER/lkp/paths:/lkp/paths \
	           -v /home/$USER/lkp/benchmarks:/lkp/benchmarks \
	           -v /home/$USER/lkp/result:/lkp/result \
	           -v /home/$USER/lkp/jobs:/lkp/jobs \
	           lkp-tests/${image}
```

**Comment by j-chiam** [comment]
Added a new issue for resolving git clone timeout
